python/cuml/dask/ensemble/randomforestclassifier.py

```diff
@@ -0,0 +1,332 @@
+                    
+        for i in range(n_workers):
+            self.n_estimators_per_worker.append(n_est_per_worker)
+            
+        remaining_est = n_estimators - (n_est_per_worker * n_workers)
+                
+        for i in range(remaining_est):
+            self.n_estimators_per_worker[i] = self.n_estimators_per_worker[i] + 1
+                    
+        ws = list(zip(workers, list(range(len(workers)))))
+                
+        self.rfs = {parse_host_port(worker):c.submit(RandomForestClassifier._func_build_rf, 
+                             n, self.n_estimators_per_worker[n], 
+                             max_depth, handle,
+                             max_features, n_bins,
+                             split_algo, split_criterion, 
+                             min_rows_per_node,
+                             bootstrap, bootstrap_features,
+                             type_model, verbose,
+                             rows_sample, max_leaves, quantile_per_tree,
+                             gdf_datatype, random.random(),
+                             workers=[worker])
+            for worker, n in ws}
+        
+        rfs_wait = list()
+        for r in self.rfs.values():
+            rfs_wait.append(r)
+                
+        wait(rfs_wait)
+        
+        
+    @staticmethod
+    def _func_build_rf(n, n_estimators, max_depth, handle,
+                             max_features, n_bins,
+                             split_algo, split_criterion, min_rows_per_node,
+                             bootstrap, bootstrap_features,
+                             type_model, verbose,
+                             rows_sample, max_leaves, quantile_per_tree,
+                             gdf_datatype, r):
+        
+        return cuRFC(n_estimators=n_estimators, max_depth=max_depth, handle=handle,
+                  max_features=max_features, n_bins=n_bins,
+                  split_algo=split_algo, split_criterion=split_criterion, min_rows_per_node=min_rows_per_node,
+                  bootstrap=bootstrap, bootstrap_features=bootstrap_features,
+                  type_model=type_model, verbose=verbose,
+                  rows_sample=rows_sample, max_leaves=max_leaves, quantile_per_tree=quantile_per_tree,
+                  gdf_datatype=gdf_datatype)
+                
+    
+    @staticmethod
+    def _fit(model, X_df, y_df, r): 
+        return model.fit(X_df, y_df)
+    
+    @staticmethod
+    def _predict(model, X, r): 
+        return model._predictGetAll(X)
+    
+    def fit(self, X, y):
+        """
+        Fit the input data to Random Forest classifier
+
+        Parameters
+        ----------
+        X : Acceptable format: dask-cudf. Dense matrix (floats or doubles) of shape (n_samples, n_features).
+        y : Acceptable format: dask-cudf. Dense matrix (floats or doubles) of shape (n_samples, 1)
+        """
+        c = default_client()
+
+        X_futures = c.sync(_extract_ddf_partitions, X)
```

Comment:
call `cuml.dask.common.dask_df_utils` Should be a direct replacement.

Reply:
Per our discussion, leaving it as it is.

Reply:
Forgive my misunderstanding here, but I didn't recall the outcome of the discussion being to leave it where it is. The only difference between the function you've created and the one in `cuml.dask.common.dask_df_utils` is that you are building a dict here, rather than a list. 

A list of 2-item tuples can be easily converted to a dict in Python using the function `dict` as demonstrated in this example:
```
>>> a = [("worker1", "part1"), ("worker2", "part2")]
>>> dict(a)
{'worker1': 'part1', 'worker2': 'part2'}
```

My recommendation to you, since this is the only delta between your `_extract_ddf_partitions` and `cuml.dask.common.utils.extract_ddf_partitions` is to use the version in `common.utils` and wrap the result in a `dict()` call.

For example:
```
        X_futures = dict(c.sync(extract_ddf_partitions, X))
```

Also, dicts are unordered when iterating through them and that's going to mean that the entries in the resulting dask array don't necessarily match the entries in the input dask array. I have not verified this yet, but I had removed the explicit cast to dict from the `extract_ddf_partitions` function because I found that I could control the order of the output partitions in order to match the order of input partitions by making sure I was always using a list.

A simple way that you could fix this is to keep the X_futures as a list of tuples and iterate through that, and cast the `y_futures` to a dict, pulling the part out for each worker encountered in each `X_futures` iteration.

Does that make sense?

Reply:
The main idea here is strictly for maintainability. We will likely need to change this `extract_ddf_partitions` at some point in the future and it's more maintainable to consolidate these changes to a single place.